samples/subsys/usb/midi/src/main.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024 Titouan Christophe
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * @file
+ * @brief Sample application for USB MIDI 2.0 device class
+ */
+
+#include <sample_usbd.h>
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/usb/class/usb_midi.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sample_usb_midi, LOG_LEVEL_INF);
+
+static const struct device *const midi = DEVICE_DT_GET(DT_NODELABEL(usb_midi));
+
+/* On boards that have a user button; button press/release -> MIDI note on/off */
+#if DT_NODE_EXISTS(DT_NODELABEL(user_button))
+static const struct device *const button = DEVICE_DT_GET(DT_PARENT(DT_NODELABEL(user_button)));
+
+static void key_press(struct input_event *evt, void *user_data)
+{
+	uint8_t command = evt->value ? MIDI_NOTE_ON : MIDI_NOTE_OFF;
+
+	usb_midi_send(midi, &UMP_MIDI1(0, command, 0, 0x40, 0x7f));
+}
+INPUT_CALLBACK_DEFINE(button, key_press, NULL);
+#endif
```

Comment:
sw0 alias is the more common/standard way of describing the "main" button on a board, so switch to that.
In fact, you may possibly look for sw1, sw2, ... too while you're at it, and play a n+1, n+2, etc. pitch when those are hit.

I would register the callback using `INPUT_CALLBACK_DEFINE(NULL, key_press, NULL);` and check keycodes in the callback (you may get the keycodes for sw0, sw1, etc. using DT_PROP macro). In fact, right now the way your callback is registered and written, it actually reacts to any button being pressed on the same gpio controller as "user_button", which is probably not what you want :)